### PR TITLE
Merge faisalman 0.7.33 and bump to 0.7.33

### PR DESCRIPTION
**Pull request by kevinyou**
# What I did
I am merging the 0.7.33 tag of the faisalman repo to the master branch of the amplitude repo
```
git remote add real git@github.com:faisalman/ua-parser-js.git
git fetch real
git pull real 0.7.33
```
To help resolve conflicts, I linted copies of https://github.com/faisalman/ua-parser-js/blob/0.7.33/src/ua-parser.js and https://github.com/amplitude/ua-parser-js/blob/master/src/ua-parser.js and then produced a diff: https://github.com/kevinyou/demo-ua-parser-comparison/blob/main/diff.diff . These include any changes to source code, including the security vulnerability and slight user agent tweaks.

Then I ran
```
npm run test-ci
```
and fixed lint and test issues (copied the device-test.json file)

Then I ran `npm run build`

### CODE REVIEW

File: src/ua-parser.js [hunk 1]
```diff
@@ -452,14 +459,6 @@
       [MODEL, [VENDOR, SAMSUNG], [TYPE, MOBILE]],
       [
         // Apple
-        /((ipod|iphone)\d+,\d+)/i // iPod/iPhone model
```
Comment (yuhao900914) [inline src/ua-parser.js:455]: Thanks for contributing to amplitude. Can you comment out this code? 
This is what we added in order to read the detail model for iphone, ipod, ipad if the model information exist. Please check the [pr](https://github.com/amplitude/ua-parser-js/commit/adbac2be284234b0cf8ff22e17b2ecec2036a8b8).

**Review by yuhao900914 — CHANGES_REQUESTED** [review]
LGTM, except for the changes of removing the device related to the iPod/iPhone model.  
Also, add my team, for more review.

File: src/ua-parser.js [hunk 2]
```diff
@@ -90,7 +91,7 @@
     },
     trim = function (str, len) {
       if (typeof str === STR_TYPE) {
-        str = str.replace(/^\s\s*/, EMPTY).replace(/\s\s*$/, EMPTY);
+        str = str.replace(/^\s\s*/, EMPTY);
```
Comment (liuyang1520) [inline src/ua-parser.js:94]: Thank you for supporting Amplitude! This change will keep the ending spaces, example:
```
"  abc  ".replace(/^\s\s*/, "").replace(/\s\s*$/, "")
// 'abc'
"  abc  ".replace(/^\s\s*/, "")
// 'abc  '
```
Not sure why, so double check here, is this change expected?

Comment (liuyang1520) [reply src/ua-parser.js:94]: Never mind, found the change for this with reason: https://github.com/faisalman/ua-parser-js/commit/a6140a17dd0300a35cfc9cff999545f267889411

File: src/ua-parser.js (hunk 1 above)
Comment (kevinyou) [reply src/ua-parser.js:455]: Reverted these - added back the capturing of iphone version and test cases

Comment (kevinyou) [reply src/ua-parser.js]: Good catch, updated

**Review by yuhao900914 — APPROVED** [review]

**Comment by kevinyou** [comment]
@yuhao900914 Looks like it's been approved. Could we get this merged?

**Pull request merged**